File: typescript/src/server/file-queue.ts
export interface PLATFileQueueRequest {
  id: string
  operationId?: string
  method: string
  path: string
  headers?: Record<string, string>
  input?: Record<string, any>
}

export interface PLATFileQueueSuccessResponse {
  id: string
  ok: true
  result: unknown
  statusCode: number
}

export interface PLATFileQueueErrorResponse {
  id: string
  ok: false
  error: {
    status?: number
    message: string
    data?: unknown
  }
}

export type PLATFileQueueResponse = PLATFileQueueSuccessResponse | PLATFileQueueErrorResponse

File: python/z2/file_queue.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass
class FileQueueRequest:
    id: str
    method: str
    path: str
    operationId: str | None = None
    headers: dict[str, str] | None = None
    input: dict[str, Any] | None = None


@dataclass
class FileQueueSuccessResponse:
    id: str
    ok: bool
    result: Any
    statusCode: int


@dataclass
class FileQueueErrorResponse:
    id: str
    ok: bool
    error: dict[str, Any]
